pkg/cluster/internal/kubeadm/config.go

```diff
@@ -158,6 +158,7 @@ nodeRegistration:
   kubeletExtraArgs:
     fail-swap-on: "false"
     node-ip: "{{ .NodeAddress }}"
+    cgroup-root: "/kubelet"
```

Comment:
setting this hardcoded breaks using older images right?

Reply:
that is something we can do, but avoiding it is preferable.
I think maybe information flows the wrong direction here.

Reply:
Sure, what did you have in mind? Do we want to try to determine the current cgroup parent from the docker daemon's configuration?

Reply:
I'm thinking maybe for changes like this that we can gracefully upgrade behavior on we could autodetect it.
Let's not block this PR on this though, I think that's a more general problem worth solving more generally.

Reply:
Sounds good, I'll do some manual testing and make sure this works (though perhaps your test suite will already catch if this breaks older images) and, if not, figure out some way to toggle it.

Reply:
Yeah, looks like it breaks older images. Do you have an idea how you want this feature gated already?

Reply:
Okay, it's not internally configurable (not exposed to the user). It gets toggled on here: https://github.com/kubernetes-sigs/kind/pull/1747/files#diff-f9926f17f93e2c4abbb007effdbe351eR56

Any ideas how we can toggle this off for legacy images?

Reply:
I moved this configuration entirely into the image by adding `--cgroup-root=/kubelet` to the kubelet flags in `images/base/files/etc/systemd/system/kubelet.service.d/10-kubeadm.conf`, so this is backwards compatible since it we're not setting the `cgroup-root` via the generated Kubeadm YAMLs and it is entirely contained in the image.

Any thoughts?